app/models/recipe.rb

```diff
@@ -1,3 +1,13 @@
 class Recipe < ApplicationRecord
+   has_many :recipe_ingredients
+   has_many :ingredients, through: :recipe_ingredients
    validates_presence_of :name, :complexity, :genre
+
+   def total_cost
+      total = 0
+      self.ingredients.each do |i|
+         total += i.cost
+      end
+      total
+   end
```

Comment:
While this method theoretically does what it is intended to do which is to get the total cost of all the ingredients for this recipe, we are using Ruby to process the data which is something we want to avoid as much as possible. 

At this point in Mod 2 it's not the expectation for you to know about aggregate functions but a possible alternative solution would be:

```ruby
def total_cost
  self.ingredients.sum(:cost) # .sum is one of the 5 aggregate methods you will learn about soon.
end
```